### PR TITLE
(PC-8863) DeeplinkImporter

### PR DESCRIPTION
**Pull request by blixit**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-8863

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -441,6 +441,8 @@ PODS:
     - React
   - RNCAsyncStorage (1.12.0):
     - React
+  - RNCClipboard (1.7.0):
```
Comment (marionnousvalentin) [inline ios/Podfile.lock:444]: tu peux enlver ça ? ça n va pas passer dans le code push ^^

Comment (marionnousvalentin) [reply ios/Podfile.lock:444]: en plus tu n dois pas l'utiliser :p

File: src/features/deeplinks/utils.test.ts [hunk 2]
```diff
@@ -108,14 +108,27 @@ describe('Formatting deeplink url', () => {
 
       expect(handleDeeplinkUrl).toBeCalledWith({ url: 'https://a.link' })
     })
-    it('should do anything for firebase short dynamic link', () => {
+    it.each([undefined, false])(
```
Comment (marionnousvalentin) [inline src/features/deeplinks/utils.test.ts:111]: ❓ ok mettre un each alors qu'il n'y a qu'un cas ?

File: src/features/deeplinks/utils.ts [hunk 3]
```diff
@@ -68,6 +69,9 @@ export const resolveHandler = (handleDeeplinkUrl: (event: DeeplinkEvent) => void
```
Comment (marionnousvalentin) [inline src/features/deeplinks/utils.ts:69]: ❓ pourquoi c'est utile ici?

File: src/features/deeplinks/pages/DeeplinkImporter.test.tsx [hunk 4]
```diff
@@ -13,4 +18,17 @@ describe('DeeplinkImporter', () => {
...
+    Clipboard.setString(url)
+
+    const resolveHandlerSpy = jest.spyOn(DeeplinkUtils, 'resolveHandler')
+
+    const { getByTestId } = render(<DeeplinkImporter />)
+
+    const importButton = getByTestId('button-container-import')
+    fireEvent.press(importButton)
+
+    resolveHandlerSpy.mockRestore()
```
Comment (marionnousvalentin) [inline src/features/deeplinks/pages/DeeplinkImporter.test.tsx:32]: ✅  tu testes quoi ici ?

File: src/features/deeplinks/utils.test.ts (hunk 2 above)
Comment (blixit) [reply src/features/deeplinks/utils.test.ts:111]: non il y a 2 cas:
- undefined, non précisé
- false, précisé mais false

File: src/features/deeplinks/utils.ts (hunk 3 above)
Comment (blixit) [reply src/features/deeplinks/utils.ts:69]: parce que c'est réutilisé à 2 endroits

File: src/features/deeplinks/pages/DeeplinkImporter.test.tsx (hunk 4 above)
Comment (blixit) [reply src/features/deeplinks/pages/DeeplinkImporter.test.tsx:32]: il manque cette ligne:
```ts
expect(resolveHandlerSpy).toBeCalled()
```

Comment (marionnousvalentin) [reply src/features/deeplinks/pages/DeeplinkImporter.test.tsx:33]: ui

File: __mocks__/@react-native-clipboard/clipboard.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+function ClipboardModule() {
+  var content = ''
+
```
Comment (marionnousvalentin) [inline __mocks__/@react-native-clipboard/clipboard.ts:3]: on pourra suppr ce fichier ^^

**Review by marionnousvalentin — APPROVED** [review]

**Pull request merged**